soc/riscv/riscv-privilege/opentitan/soc.c

```diff
@@ -8,10 +8,18 @@
 #include <zephyr/devicetree.h>
 #include <zephyr/init.h>
 
+#define PWRMGR_BASE (DT_REG_ADDR(DT_NODELABEL(pwrmgr)))
 #define RV_TIMER_BASE (DT_REG_ADDR(DT_NODELABEL(mtimer)))
 
 static int soc_opentitan_init(void)
 {
+	/* Enable the watchdog reset (bit 1). */
+	sys_write32(2u, PWRMGR_BASE + PWRMGR_RESET_EN_REG_OFFSET);
+	/* Write CFG_CDC_SYNC to commit change. */
+	sys_write32(1u, PWRMGR_BASE + PWRMGR_CFG_CDC_SYNC_REG_OFFSET);
+	/* Poll CFG_CDC_SYNC register until it reads 0. */
+	while (sys_read32(PWRMGR_BASE + PWRMGR_CFG_CDC_SYNC_REG_OFFSET)) {
+	}
```

Comment:
question: why isn't this handled by the driver itself?

Reply:
Adding the watchdog to the SoC's list of enabled reset sources seems more appropriate to be in the SoC init code.